File: examples/2/2.1/simple.js
exports.__esModule = true;
exports.getElementsValue = getElementsValue
exports.startTracing = startTracing
exports.stopTracing = stopTracing

async function startTracing(context) {
    await context.tracing.start({screenshots: true, snapshots: true});
}

async function stopTracing(tracefile, context, logger) {
    logger(`Tracing file: ${tracefile}`);
    await context.tracing.stop({ path: tracefile });
}


async function getElementsValue(selector, page, logger) {
    let elements = await page.locator(selector).all();
    const values = [];
    for (let element of elements) {
        let value = await element.inputValue();
        console.log(`console: Selector: ${selector} Value: ${value}`);
        await page.evaluate((arg) => console.log(`page: Selector: ${arg[0]} Value: ${arg[1]}`), [selector, value]);
        logger(`logger: Selector: ${selector} Value: ${value}`);
        values.push(value);
    }
    return values;
}
getElementsValue.rfdoc = "This keyword is implemented in JS. It gets the value of the elements."

File: examples/2/2.1/simple.py
from pathlib import Path

from Browser import Browser  # type: ignore
from Browser.base.librarycomponent import LibraryComponent  # type: ignore
from robot.api import logger  # type: ignore
from robot.api.deco import keyword


class simple(LibraryComponent):
    def __init__(self, library: Browser):
        super().__init__(library)
        self.initialize_js_extension(Path(__file__).parent.resolve() / "simple.js")

    @keyword
    def get_elements_value(self, selector: str) -> list[str]:
        """Get elements value."""
        selector = self.resolve_selector(selector)
        values = list(self.call_js_keyword("getElementsValue", selector=selector))
        logger.info(f"Values: {values}")
        return values

    @keyword
    def start_tracing(self):
        """Start tracing."""
        self.call_js_keyword("startTracing")

    @keyword
    def stop_tracing(self, path: Path):
        """Stop tracing."""
        return self.call_js_keyword("stopTracing", tracefile=str(path.resolve().absolute()))
